Source/WebCore/dom/ShadowRoot.h

```diff
@@ -181,6 +186,8 @@ class ShadowRoot final : public DocumentFragment, public TreeScope {
     std::unique_ptr<Style::Scope> m_styleScope;
     std::unique_ptr<SlotAssignment> m_slotAssignment;
     mutable std::optional<PartMappings> m_partMappings;
+    
+    AtomString m_referenceTarget;
```

Comment:
This is potentially controversial to add to every shadow root for a feature likely to be relatively seldom used. Is there any good alternative?

Reply:
WebKit has something called `NodeRareData` for this scenario. But it may be good for someone who is more knowledgeable about WebKit's DOM code to weigh in on whether that's appropriate. Maybe transfer this comment to your full-fledged review PR?